test/math.jl

```diff
@@ -842,24 +842,103 @@ end
         @test atan2(-T(1235.2341234), -T(2.5)) === -(T(pi)-(atan(abs(-T(1235.2341234)/T(2.5)))-_ATAN2_PI_LO(T)))
     end
 end
+#prev, current, next float
+pcnfloat(x) = prevfloat(x), x, nextfloat(x)
+@testset "sinh" begin
+    for T in (Float32, Float64)
+        @test sinh(zero(T)) === zero(T)
+        @test sinh(-zero(T)) === -zero(T)
+        @test sinh(nextfloat(zero(T))) === nextfloat(zero(T))
+        @test sinh(prevfloat(zero(T))) === prevfloat(zero(T))
+        @test sinh(T(1000)) === T(Inf)
+        @test sinh(-T(1000)) === -T(Inf)
+        @test isnan_type(T, sinh(T(NaN)))
+    end
+    for x in (pcnfloat(2.0^-28)..., pcnfloat(22.0)..., pcnfloat(709.7822265633563)...)
```

Comment:
Seems like it would be better to use `SINH_SMALL_X(T)`, `H_MEDIUM_X(T)`, etcetera (importing those symbols into the test).  That way it will unify the tests for both `T`, make it clearer where these thresholds come from, and ensure the tests are updated if the thresholds change (or if the `sinh` code is rewritten).